lib/build-es-document.js

```diff
@@ -46,17 +43,37 @@ const buildEsDocument = async ({ type, records }) => {
 
 const buildSierraBibs = (records) => {
   // instantiate sierra bib per bib record
-  const bibs = records.map((record) => new SierraBib(record))
-  // add bibs to holding and item records on bibs
-  bibs.forEach((bib) => {
-    if (bib.items()) {
-      bib.items().forEach((i) => i.addBib(bib))
-    }
-    if (bib.holdings()) {
-      bib.holdings().forEach((h) => h.addBib(bib))
-    }
-  })
-  return bibs
+  return records
+    .map((record) => new SierraBib(record))
+    .map((bib) => {
+      // Ensure each holding has reverse reference to bib:
+      bib._holdings = bib._holdings.map((holding) => {
+        holding._bibs = [bib]
+        return holding
+      })
+      // Ensure each item has reverse reference to bib:
+      bib._items = bib._items.map((item) => {
+        item._bibs = [bib]
+        return item
+      })
+      return bib
+    })
+}
+
+/**
+ *  Given an event type (e.g. Bib, Item, Holding) and an array of plainobject records,
+ *  returns an array of plainobjects representing the relevant Sierra bibs.
+ */
+const transformIntoBibRecords = async (type, records) => {
+  // If incoming event is Item/Holding, convert it into a Bib event:
+  if (type === 'Item') {
+    records = await filteredSierraItemsForItems(records)
+    records = await platformApi.bibsForHoldingsOrItems(type, records)
+  } else if (type === 'Holding') {
+    records = await filteredSierraHoldingsForHoldings(records)
+    records = await platformApi.bibsForHoldingsOrItems(type, records)
+  }
+  return records
```

Comment:
is there a chance that records may contain duplicate bibs we'd want to filter out?

Reply:
Good idea. I'll make sure that's happening and add some test coverage

Reply:
This is really controlled by `_bibIdentifiersForHoldings` and `_bibIdentifiersForItems`. I updated the former to ensure it de-dupes bib identifiers. The latter was already de-duping, but I cleaned it up a little. Added specific tests for both.